### DISCUSSION TITLE
Lightning CSS

### DISCUSSION THREAD

**Discussion by ecatugy** [mantinedev/mantine#6253] [Feature requests]
Why aren’t we leveraging Lightning CSS instead of PostCSS? We’d see a substantial performance increase given that Lightning CSS’s engine is built on Rust, not JavaScript like PostCSS. Just look at the latest TailwindCSS version, they’ve migrated from PostCSS to Lightning CSS and the speed enhancement is significant!

**Comment by rtivital** [comment]
I do not see any performance issues with postcss, it works fast enough.

**Reply by ecatugy** [reply]
Alright, I didn’t say PostCSS has performance issues, my suggestion’s a potential improvement in that regard, and I’ve put forth my arguments. Some big players are already migrating, I know it’s pretty tough to migrate, and y’all recently switched from emotion to PostCSS and a lot of folks complained, but it’s part of the process, changes always happen or you get left behind. The trend is for frontend tools to increasingly use resources made in Rust, this is a trend that seems inevitable in the long run.

**Reply by rtivital** [reply]
So the tool that is currently being used is fine – it is performant enough for its task and does its job well. But we should migrate to another hyped tool and introduce breaking changes because other people did that. Is this correct?

**Reply by ecatugy** [reply]
It was a suggestion for y’all to keep on your radar for the future. Tools migrating to Rust isn’t hype, proof of this is that Next itself migrated the compiler to Rust starting from version 12. If Lightning CSS is in the hype, okay, it might be, but tools being migrated to an engine in Rust isn’t hype, it’s a reality. That’s why I made my suggestion for the future.
